### PR TITLE
use `limit_zero` macro instead of `limit 0`

### PR DESCRIPTION
**Pull request by swanderz**
This is a:
- [x] bug fix PR with no breaking changes — please ensure the base branch is `master`
- [ ] new functionality — please ensure the base branch is the latest `dev/` branch
- [ ] a breaking change — please ensure the base branch is the latest `dev/` branch

## Description & motivation

This test cannot be used by databases that don't support `limit zero`

## Checklist
- [ ] I have verified that these changes work locally on the following warehouses (Note: it's okay if you do not have access to all warehouses, this helps us understand what has been covered)
    - [ ] BigQuery
    - [ ] Postgres
    - [ ] Redshift
    - [ ] Snowflake
- [x] I have "dispatched" any new macro(s) so non-core adapters can also use them (e.g. [the `star()` source](https://github.com/fishtown-analytics/dbt-utils/blob/master/macros/sql/star.sql))
- [x] I have updated the README.md (if applicable)
- [ ] I have added tests & descriptions to my models (and macros if applicable)
- [ ] I have added an entry to CHANGELOG.md

### CODE REVIEW

File: integration_tests/tests/jinja_helpers/test_slugify.sql [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {% if dbt_utils.slugify('!Hell0 world-hi') == 'hell0_world_hi' %}
     {# Return 0 rows for the test to pass #}
-    select 1 limit 0
+    select 1 {{ limit_zero }}
```
Comment (joellabes) [inline integration_tests/tests/jinja_helpers/test_slugify.sql:3]: the error doesn't lend itself to this being the problem, but I noticed that the other places that use {{ limit_zero }} do alias the column so let's give that a go. 

```suggestion
    select 1 as col_name {{ limit_zero }}
```

**Review by joellabes — CHANGES_REQUESTED** [review]
The CI test is failing with 

```
Database Error in test test_slugify (tests/jinja_helpers/test_slugify.sql)
  syntax error at or near "object"
  LINE 9:     select 1 <dbt.clients.jinja.MacroGenerator object at 0x7...
                                                         ^
  compiled SQL at target/run/dbt_utils_integration_tests/tests/jinja_helpers/test_slugify.sql
Encountered an error:
FailFast Error in test test_slugify (tests/jinja_helpers/test_slugify.sql)
  Failing early due to test failure or runtime error
```

However a test like `assert_pretty_time_is_string` _is_ passing and looks almost identical. The only thing that jumps out is this one doesn't alias `select 1`. It shouldn't _need_ to - the code you've written works for me when I test it in dbt Cloud - but let's give it a whirl just in case.

File: integration_tests/tests/jinja_helpers/test_slugify.sql [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {% if dbt_utils.slugify('!Hell0 world-hi') == 'hell0_world_hi' %}
     {# Return 0 rows for the test to pass #}
-    select 1 limit 0
+    select 1 as col_name {{ limit_zero }}
```
Comment (joellabes) [inline integration_tests/tests/jinja_helpers/test_slugify.sql:3]:
```suggestion
    select 1 as col_name {{ limit_zero() }}
```

**Review by joellabes — CHANGES_REQUESTED** [review]
It's a macro! that's why there was an error 🎉

**Comment by swanderz** [comment]
thanks @joellabes this should be ready now IMHO

**Review by joellabes — APPROVED** [review]
Thanks as always for your contribution!

**Pull request merged**